packages/typescript-estree/src/create-program/describeFilePath.ts

```diff
@@ -0,0 +1,31 @@
+import path from 'path';
+
+export function describeFilePath(
+  filePath: string,
+  tsconfigRootDir: string,
+): string {
+  // If the TSConfig root dir is a parent of the filePath, use
+  // `<tsconfigRootDir>` as a prefix for the path.
+  const relative = path.relative(tsconfigRootDir, filePath);
+  if (relative && !relative.startsWith('..') && !path.isAbsolute(relative)) {
+    return `<tsconfigRootDir>/${relative}`;
+  }
+
+  // Root-like Mac/Linux (~/*, ~*) or Windows (C:/*, /) paths that aren't
+  // relative to the TSConfig root dir should be fully described.
+  // This avoids strings like <tsconfigRootDir>/../../../../repo/file.ts.
+  // https://github.com/typescript-eslint/typescript-eslint/issues/6289
+  if (/^[(\w+:)\\/~]/.test(filePath)) {
```

Comment:
```ts
/^[(\w+:)\\/~]/.test('test.ts');
/^[(\w+:)\\/~]/.test('test/test/test.ts');
```
this regexp seem to fail when path starts with string

Reply:
Yeah - my thought was to make the `<tsconfigRootDir>` prefix used only as a shorthand to reduce long paths.